kani-compiler/src/kani_middle/transform/check_aliasing/actions.rs

```diff
@@ -0,0 +1,237 @@
+            Rvalue::Discriminant(_) => {}
+            Rvalue::Len(_) => {}
+            Rvalue::Repeat(_, _) => {}
+            Rvalue::ShallowInitBox(_, _) => {}
+            Rvalue::ThreadLocalRef(_) => {}
+            Rvalue::NullaryOp(_, _) => {}
+            Rvalue::UnaryOp(_, _) => {}
+            Rvalue::Use(_) => {}
+        }
+    }
+
+    /// Collect the actions for the places of a statement
+    fn visit_statement_places(&mut self, stmt: &Statement) {
+        match &stmt.kind {
+            StatementKind::Assign(place, rvalue) => {
+                self.visit_rvalue_places(rvalue);
+                self.visit_place(place);
+            }
+            StatementKind::FakeRead(_, _) => {}
+            StatementKind::SetDiscriminant { .. } => {}
+            StatementKind::Deinit(_) => {}
+            StatementKind::StorageLive(_) => {}
+            StatementKind::StorageDead(_) => {}
+            StatementKind::Retag(_, _) => {}
+            StatementKind::PlaceMention(_) => {}
+            StatementKind::AscribeUserType { .. } => {}
+            StatementKind::Coverage(_) => {}
+            StatementKind::Intrinsic(_) => {}
+            StatementKind::ConstEvalCounter => {}
+            StatementKind::Nop => {}
+        }
+    }
+
+    /// Collect the actions for the places of the statement,
+    /// then find assignments of pointer values to lvalues
+    /// and collect updates to the stacked borrows state
+    /// accordingly
+    pub fn visit_statement(&mut self, stmt: &Statement) {
+        self.visit_statement_places(stmt);
+        match &stmt.kind {
+            StatementKind::Assign(to, rvalue) => {
+                match rvalue {
+                    Rvalue::Ref(_, BorrowKind::Mut { .. }, from) => {
+                        self.visit_assign_reference(to.local, from.clone());
+                    }
+                    Rvalue::AddressOf(Mutability::Mut, from) => {
+                        self.visit_assign_pointer(to.local, from.clone());
+                    }
+                    Rvalue::Use(Operand::Constant(_)) => {
+                        // Do nothing for the constants case
+                    }
+                    Rvalue::Use(Operand::Copy(_)) => {
+                        eprintln!("Copy not yet handled");
+                        // Do nothing for the constants case
+                    }
+                    Rvalue::Use(Operand::Move(_)) => {
+                        eprintln!("Move not yet handled");
+                        // Do nothing for the constants case
+                    }
+                    Rvalue::BinaryOp(_, _, _) => {
+                        eprintln!("Binary op not yet handled");
+                    }
+                    Rvalue::CheckedBinaryOp(_, _, _) => {
+                        eprintln!("Checked binary op not yet handled");
+                    }
+                    _ => {
+                        panic!("Rvalue kind: {:?} not yet handled", rvalue);
```

Comment:
Is there any reason why this is a panic, unlike other variants?

Reply:
This is the fall-through case; the other variants occur in the "control_flow.rs" file, but don't alter the stacked borrows state.